From b478960ce19f469e46e4014c72103bb534f8741f Mon Sep 17 00:00:00 2001
From: Tim Lievens <tim.lievens@heilig-graf.be>
Date: Thu, 21 Feb 2019 11:05:11 +0100
Subject: [PATCH 1/1] Disabled access to password form

---
 openid_connect.module | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/openid_connect.module b/openid_connect.module
index c8c1b93..cab983e 100644
--- a/openid_connect.module
+++ b/openid_connect.module
@@ -471,7 +471,7 @@ function openid_connect_form_user_profile_form_alter(&$form, &$form_state) {
     $account_form = &$form;
   }
 
-  if (!empty($account_form['pass']['#access']) && !openid_connect_set_password_access($form['#user'])) {
+  if (!empty($account_form['current_pass']['#access']) && !openid_connect_set_password_access($form['#user'])) {
     $account_form['current_pass']['#access'] = FALSE;
     $account_form['current_pass_required_values']['#value'] = array();
     $account_form['pass']['#access'] = FALSE;
@@ -817,6 +817,7 @@ function openid_connect_complete_authorization($client, $tokens, &$destination)
   // Deny access if any module returns FALSE.
   if (in_array(FALSE, $results, TRUE)) {
     watchdog('openid_connect', 'Login denied for @email via pre-login hook.', array('@email' => $userinfo['email']), WATCHDOG_ERROR);
+    drupal_set_message(t('Login with Office365 denied for @mail, use normal login form.', array('@mail' => $userinfo['email'])), 'error');
     return FALSE;
   }
 
-- 
2.12.0.windows.1

